src/Search/Searchable.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace DoubleThreeDigital\Runway\Search;
+
+use DoubleThreeDigital\Runway\AugmentedRecord;
+use DoubleThreeDigital\Runway\Runway;
+use Statamic\Contracts\Data\Augmentable;
+use Statamic\Contracts\Data\Augmented;
+use Statamic\Contracts\Query\ContainsQueryableValues;
+use Statamic\Contracts\Search\Result;
+use Statamic\Contracts\Search\Searchable as Contract;
+use Statamic\Data\HasAugmentedInstance;
+use Statamic\Facades\Site;
+use Statamic\Search\Result as ResultInstance;
+
+class Searchable implements Contract, ContainsQueryableValues, Augmentable
+{
+    use HasAugmentedInstance;
+
+    protected $model;
+    protected $resource;
+
+    public function __construct($model)
+    {
+        $this->model = $model;
+        $this->resource = Runway::findResourceByModel($model);
+    }
+
+    public function resource()
+    {
+        return $this->resource;
+    }
+
+    public function getQueryableValue(string $field)
+    {
+        if ($field === 'site') {
+            return Site::current()->handle();
+        }
+
+        return $this->model->{$field};
+    }
+
+    public function getSearchValue(string $field)
+    {
+        return $this->model->{$field};
+    }
+
+    public function getSearchReference(): string
+    {
+        return vsprintf('runway::%s::%s', [
+            $this->resource->handle(),
+            $this->model->getKey(),
+        ]);
+    }
+
+    public function toSearchResult(): Result
+    {
+        return new ResultInstance($this, 'runway:'.$this->resource->handle());
```

Comment:
The second argument here is what you get when you use `{{ result_type }}` on the frontend.

Entries would be `entry`, terms would be `term`, etc.

Here I've used `runway:resourcehandle` so it wouldn't conflict if someone else was to have something called `resourcehandle`.

Maybe that's not important to you so you could remove the prefix, or maybe you want to change it to `runway_resourcehandle`.

Reply:
Thanks! I'm good with the `runway:` prefix 👍